Classes/Private/Manager/Video/OCTVideoEngine.m

```diff
@@ -125,22 +125,24 @@ - (BOOL)isSendingVideo
     return self.captureSession.isRunning;
 }
 
-- (CALayer *)videoCallPreview
+- (void)getVideoCallPreview:(void (^)(CALayer *))completionBlock
 {
     DDLogVerbose(@"%@: videoCallPreview", self);
-    AVCaptureVideoPreviewLayer *previewLayer = [AVCaptureVideoPreviewLayer layerWithSession:self.captureSession];
-
-    return previewLayer;
+    dispatch_async(self.processingQueue, ^{
+        AVCaptureVideoPreviewLayer *previewLayer = [AVCaptureVideoPreviewLayer layerWithSession:self.captureSession];
+        completionBlock(previewLayer);
+    });
 }
 
-- (UIView *)videoFeedWithRect:(CGRect)rect;
+- (void)getVideoFeed:(void (^)(UIView *videoFeed))completionBlock
 {
-    DDLogVerbose(@"%@: videoFeedWithRect", self);
-    if (! self.videoView) {
-        self.videoView = [[OCTVideoView alloc] initWithFrame:rect];
-    }
-
-    return self.videoView;
+    DDLogVerbose(@"%@: getVideoFeed", self);
+    dispatch_async(self.processingQueue, ^{
+        if (! self.videoView) {
+            self.videoView = [[OCTVideoView alloc] initWithFrame:CGRectZero];
+        }
+        completionBlock(self.videoView);
```

Comment:
Hmm I'll try the above and see if it makes a difference. Right now this is what Antidote looks like when setting up the video screens. Which yea, looks kind of ugly?

        UIView *loadingView = [UIView new];
        loadingView.backgroundColor = [UIColor grayColor];
        controller.videoView = loadingView;

        __weak ActiveCallViewController *weakActiveVC = controller;
        [self.manager getVideoFeed:^(UIView *videoFeed) {
            ActiveCallViewController *strongActiveVC = weakActiveVC;

            dispatch_async(dispatch_get_main_queue(), ^{
                strongActiveVC.videoView = videoFeed;
            });
        }];

        [self.manager getVideoCallPreview:^(CALayer *layer) {
            ActiveCallViewController *strongActiveVC = weakActiveVC;
            dispatch_async(dispatch_get_main_queue(), ^{
                strongActiveVC.previewLayer = layer;
            });
        }];